### PR TITLE
[#49144] Fix missing link to Meeting location on index pages  

### PR DESCRIPTION
**Pull request by aaron-contreras**
## Bug description
There just wasn't a link being rendered from the get-go, instead it was `html_escaped` plaintext.

## Demo screenshot
image

## Notes
See Bug Work Package: https://community.openproject.org/work_packages/49144

### CODE REVIEW

**Review by cbliard — CHANGES_REQUESTED** [review]
It works, but I'm not sure that's correct. A location can be a url, but it can also be a place description, a city, a room name, etc. So it should be converted to a link only if it's a valid URL.

**Comment by aaron-contreras** [comment]
@cbliard Thanks for the catch 👍. Didn't really think of using the Location field as something other than a URL. I've addressed the comment in https://github.com/opf/openproject/pull/13440/commits/7df7ec97a196d56c5a1b794e9814dc8809999736 and https://github.com/opf/openproject/pull/13440/commits/8ac6616f1afdd354252de48690e81d3ed622b27b

File: modules/meeting/app/components/meetings/row_component.rb [hunk 1]
```diff
@@ -47,7 +47,24 @@ def duration
     end
 
     def location
-      h(model.location)
+      if valid_meeting_location_link?
```
Comment (oliverguenther) [inline modules/meeting/app/components/meetings/row_component.rb]: We already use riku for automatically linking to URLs. I think we could just reuse this here? https://github.com/opf/openproject/blob/dev/lib/open_project/text_formatting/filters/autolink_filter.rb#L46

Comment (aaron-contreras) [reply modules/meeting/app/components/meetings/row_component.rb]: Thank you! Determining valid URLs turned out to be more daunting than expected. Addressed here: https://github.com/opf/openproject/pull/13440/commits/bb126714ddbb457401a0e6e2e30cb076339185e7 and a precautionary spec here https://github.com/opf/openproject/pull/13440/commits/6463d44f39bef97125a8dc1bda6dd77b958ff0ce

**Review by oliverguenther — APPROVED** [review]
Thanks for deep diving into the autolinking 🙇

**Pull request merged**